app/javascript/article-form/components/__tests__/ArticleCoverImage.test.jsx

```diff
@@ -9,6 +9,7 @@ import fetch from 'jest-fetch-mock';
 import '@testing-library/jest-dom';
 import { ArticleCoverImage } from '../ArticleCoverImage';
 
+jest.mock('react-inlinesvg');
```

Comment:
@aitchiss, @ludwiczakpawel, this fixes the errors in the tests. The problem is the react-inlinesvg package loads the src props asynchronously, so you need a server running which is not the case for this kind of test.

Consider using  [babel-plugin-inline-react-svg](https://github.com/airbnb/babel-plugin-inline-react-svg#readme) and then we won't need to mock this module, and the SVG can be imported In an easier manner, e.g.

````jsx
import AwesomeIcon from './awesome.svg';

...

const SomeComponent = () => {
	return <AwesomeIcon />
};
```

To use this, we'll need to add the plugin to our babel configuration, but that's pretty much it.

Reply:
Thanks for looking into this! I saw that the tests would pass if we mocked the lib but tbh I wasn't comfortable doing it because I couldn't understand why this lib would be preventing change events firing in this one subset of tests (the Icon component isn't used in this feature, so it feels pretty weird).

If `babel-plugin-inline-react-svg` solves the issue I definitely +1 adding it 🙌

Reply:
Oh maaan, sooooo good, thanks!! 

Re babel alternative - I'll give it a look.

Reply:
I've tried that one, and it caused some issues in both storybook and product itself, so if that's ok with you, I'd rather stick to `react-inlinesvg` for now, unless you wanna give it a try?

Reply:
Yeah mocking is not ideal as the icon will literally not render. Also I missed up top that you had tried this @aitchiss.

What issues did you encounter with the Babel plugin @ludwiczakpawel? I've used it before, although admittedly on a small side project only.

Reply:
> Yeah mocking is not ideal as the icon will literally not render

I'm also thinking this issue could randomly pop up again in future and it's not going to be the first thing on the dev's mind to mock this package for that test, especially if the component isn't even using it 🙃  (I guess we could mock globally, but like you say it's not ideal either).

Reply:
> What issues did you encounter with the Babel plugin

My local env immediately stopped working, after installing it - like it had conflicts with every other SVG we currently import elsewhere. And I had troubles turning it into "reusable" component I have right now (`<Icon />`).

I absolutely agree that this issue may pop up again in future so it's not ideal. Anyway, both of these issue I described above could be caused by me doing something wrong. If you wanna give it another look, go for it! I'd rather wait with shipping this if we can make it more bulletproof first.